scripts/provisioning/configcmd.py

```diff
@@ -30,11 +33,119 @@ def __init__(self, config: str):
     """Constructor."""
     try:
       super(ConfigCmd, self).__init__(config)
+
+      self.read_cluster_id()
+      self.read_ldap_credentials()
+      self.read_node_info()
+
+    except IOError as ioe:
+      sys.stderr.write(f'failed to open config file: {self.s3_prov_config}, err: {ioe}\n')
+      raise ioe
+    except YAMLError as ye:
+      sys.stderr.write(f'yaml load failed for file: {self.s3_prov_config}, err: {ye}\n')
+      raise ye
     except Exception as e:
+      sys.stderr.write(f'unknown exception: {e}\n')
       raise e
 
   def process(self):
     """Main processing function."""
     retval = 0
     sys.stdout.write(f"Processing {self.name} {self.url}\n")
+
+    # Configure openldap and ldap-replication
+    try:
+      retval = self.configure_openldap()
+    except Exception as e:
+      sys.stderr.write(f'In self.configure_openldap(), exception: {e}\n')
+      return 1
+
+    if retval == 0:
+      sys.stdout.write("INFO: Successfully configured openldap on the node.\n")
+      # Configure haproxy
+      try:
+        retval = self.configure_haproxy()
+      except Exception as e:
+        sys.stderr.write(f'In self.configure_haproxy(), exception: {e}\n')
+        return 1
+      if retval == 0:
+        sys.stdout.write("INFO: Successfully configured haproxy on the node.\n")
+      else:
+        sys.stderr.write("INFO: Failed to configure haproxy on the node.\n")
+        return retval
+
+    else:
+      sys.stderr.write("INFO: Failed to configure openldap on the node.\n")
+
     return retval
+
+  def configure_openldap(self):
+    """Install and Configure Openldap over Non-SSL."""
+    # 1. Install and Configure Openldap over Non-SSL.
+    # 2. Enable slapd logging in rsyslog config
+    # 3. Set openldap-replication
+    # 4. Check number of nodes in the cluster
+    cmd = ['/opt/seagate/cortx/s3/install/ldap/setup_ldap.sh',
+           '--ldapadminpasswd',
+           f'{self.ldap_passwd}',
+           '--rootdnpasswd', 
+           f'{self.rootdn_passwd}',
+           '--skipssl']
+    handler = SimpleProcess(cmd)
+    stdout, stderr, retcode = handler.run()
```

Comment:
I would suggest to use exceptions instead of return values. refer SubProcess.run() documentation :
https://docs.python.org/3/library/subprocess.html

_If check is true, and the process exits with a non-zero exit code, a CalledProcessError exception will be raised. Attributes of that exception hold the arguments, the exit code, and stdout and stderr if they were captured._

Reply:
To address this, I'll need to add changes in cortx-utils repo. If instead I try to directly use SubProcess() APIs in this code, codacy will raise warnings as it is against using subprocess(). Hence pls let me know if it is ok to go with this change, otherwise I'll need to address this in cortx-utils repo.

Reply:
ohh sorry.. I confused cortx-utils.SimpleProcess with SubProcess.

I guess we need to live with return codes, until we change the setup_ldap.sh to python also.

Reply:
As discussed, used custom exception to trigger in such failure cases. Also tested negative situations using manual fault injection in code and verified they are working.